### PR TITLE
Update BERT model file structure

### PR DESCRIPTION
**Pull request by mattdangerw**
This PR is just moving file structure and imports for BERT.

There are some code changes that should follow this up, but wanted to keep this purely organizational.

https://github.com/keras-team/keras-nlp/issues/365

### CODE REVIEW

File: keras_nlp/models/__init__.py [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
 
-from keras_nlp.models.bert import BertBase
-from keras_nlp.models.bert import BertCustom
-from keras_nlp.models.bert import BertLarge
-from keras_nlp.models.bert import BertMedium
-from keras_nlp.models.bert import BertPreprocessor
-from keras_nlp.models.bert import BertSmall
-from keras_nlp.models.bert import BertTiny
-from keras_nlp.models.bert_tasks import BertClassifier
+from keras_nlp.models.bert.bert_models import BertBase
```
Comment (jbischof) [inline keras_nlp/models/__init__.py:15]: Why change these symbols? I thought we are just reshuffling files behind the scenes

File: keras_nlp/models/bert/__init__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2022 The KerasNLP Authors
```
Comment (jbischof) [inline keras_nlp/models/bert/__init__.py:1]: Do we need this file?

File: keras_nlp/network_tests/test_load_ckpts.py [hunk 3]
```diff
@@ -16,20 +16,18 @@
...
 
 
 @pytest.mark.slow
 class BertCkptTest(tf.test.TestCase):
     def test_load_bert(self):
-        for checkpoint in bert_checkpoints:
-            bert_variant = bert_checkpoints[checkpoint]["model"]
-            bert_class = bert_model_class_by_name(bert_variant)
+        for checkpoint in bert_checkpoints.checkpoints:
+            bert_variant = bert_checkpoints.checkpoints[checkpoint]["model"]
```
Comment (jbischof) [inline keras_nlp/network_tests/test_load_ckpts.py:29]: A little clunky but perhaps inevitable

File: keras_nlp/models/bert/__init__.py (hunk 2 above)
Comment (mattdangerw) [reply keras_nlp/models/bert/__init__.py:1]: Well in a strict sense, yes, I think every directory should have an `__init__.py`. But I will make this empty.

We should have some discussion about whether we want a `keras.models.bert` documented namespace. But let's not solve any of that here. Code organization only!

File: keras_nlp/network_tests/test_load_ckpts.py (hunk 3 above)
Comment (mattdangerw) [reply keras_nlp/network_tests/test_load_ckpts.py:29]: We could consider just wrapping all the direct structs with some sort of readable function names (unexposed to user).

`bert_checkpoints.list_checkpoints_for_arch()` or `bert_checkpoints.get_checkpoint("bert_based_uncase_en")`.

But definitely not on this PR. And probably after we resolve a little more on the config specification questions.

**Review by jbischof — APPROVED** [review]
Looks good, thanks!

**Pull request merged**